primitives/phragmen/src/lib.rs

```diff
@@ -668,28 +665,22 @@ pub fn equalize<Balance, AccountId, C, FS>(
 			if diff > max_diff { max_diff = diff; }
 		}
 
-		if max_diff < tolerance {
-			break;
+		i += 1;
+		if max_diff <= tolerance || i == iterations {
```

Comment:
This is slightly safer
```suggestion
		if max_diff <= tolerance || i >= iterations {
```